code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm

```diff
@@ -439,3 +439,86 @@
 	if (get_dist(owner, receiver) > 7)
 		// Out of screen transfer.
 		owner.balloon_alert(owner, "Transferred plasma")
+
+
+#define BULWARK_LOOP_TIME 1 SECONDS
+#define BULWARK_RADIUS 4
+#define BULWARK_ARMOR_MULTIPLIER 1.25
+
+/datum/action/xeno_action/bulwark
+	name = "Royal Bulwark"
+	action_icon_state = "bulwark"
+	desc = "Creates a field of defensive energy, filling chinks in the armor of nearby sisters, making them more resilient."
+	plasma_cost = 100
+	cooldown_timer = 20 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_QUEEN_HIVE_MESSAGE,
+	)
+	/// assoc list xeno = armor_diff
+	var/list/armor_mod_keys = list()
+
+/datum/action/xeno_action/bulwark/action_activate()
+	var/list/turf/affected_turfs = RANGE_TURFS(BULWARK_RADIUS, owner)
+	add_cooldown()
+
+	for(var/turf/target AS in affected_turfs)
+		//yes I realize this adds and removes it every move but its simple
+		//also we use this and not aura because we want speedy updates on entering
+		RegisterSignal(target, COMSIG_ATOM_EXITED, PROC_REF(remove_buff))
+		RegisterSignal(target, COMSIG_ATOM_ENTERED, PROC_REF(apply_buff))
+		ADD_TRAIT(target, TRAIT_BULWARKED_TURF, XENO_TRAIT)
+		for(var/mob/living/carbon/xenomorph/xeno in target)
+			apply_buff(null, xeno)
+
+	var/obj/effect/abstract/particle_holder/aoe_particles = new(owner.loc, /particles/bulwark_aoe)
+	aoe_particles.particles.position = generator(GEN_SQUARE, 0, 16 + (BULWARK_RADIUS-1)*32, LINEAR_RAND)
+	while(do_after(owner, BULWARK_LOOP_TIME, BUSY_ICON_MEDICAL, extra_checks = CALLBACK(src, TYPE_PROC_REF(/datum/action, can_use_action), FALSE, XACT_IGNORE_COOLDOWN|XACT_USE_BUSY)))
+		succeed_activate()
+
+	aoe_particles.particles.spawning = 0
+	QDEL_IN(aoe_particles, 4 SECONDS)
+
+	for(var/turf/target AS in affected_turfs)
+		UnregisterSignal(target, list(COMSIG_ATOM_EXITED, COMSIG_ATOM_ENTERED))
+		REMOVE_TRAIT(target, TRAIT_BULWARKED_TURF, XENO_TRAIT)
+		for(var/mob/living/carbon/xenomorph/xeno AS in armor_mod_keys)
+			remove_buff(null, xeno)
+	affected_turfs = null
+
+///adds buff to xenos
+/datum/action/xeno_action/bulwark/proc/apply_buff(datum/source, mob/living/carbon/xenomorph/xeno, direction)
+	SIGNAL_HANDLER
+	if(!isxeno(xeno) || armor_mod_keys[xeno] || !owner.issamexenohive(xeno))
```

Comment:
Should the first part be an istype check? Does it matter?

Reply:
Yea, entered signal handler